modules/appeals_api/app/controllers/appeals_api/v1/decision_reviews/notice_of_disagreements/evidence_submissions_controller.rb

```diff
@@ -64,6 +72,14 @@ def submission_attributes
           }
         end
 
+        def request_headers
+          HEADERS.reduce({}) do |acc, header_key|
+            header_value = request.headers[header_key]
+
+            header_value.nil? ? acc : acc.merge({ header_key => header_value })
+          end
```

Comment:
I felt like I had seen this done a different, more succinct way - finally found it in BaseContestableIssuesController!
```suggestion
HEADERS.index_with { |key| request.headers[key] }.compact
```
`index_with` was added in Rails 6: https://www.bigbinary.com/blog/rails-6-adds-enumerable-index_with